src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -608,8 +611,19 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
           .map(cleaned => logger.info("Done cleaning {} accounts checkpoint rows.", cleaned))
       }
 
+      //if needed, we get the stored levels and only keep updates that are more recent
+      def prunedUpdates(): Future[Map[AccountId, BlockReference]] =
+        if (onlyProcessLatest) db.run {
+          val selection = ids.map(_._1).toSet
```

Comment:
you can replace `ids.map(_._1).toSet` with `ids.keySet`

Reply:
yes!

Reply:
yup!